src/fastmcp/server/auth/cimd.py

```
@@ -0,0 +1,634 @@
+        import json
+
+        from authlib.jose import JsonWebKey
+
+        # Extract kid from token header
+        try:
+            header_b64 = token.split(".")[0]
+            header_b64 += "=" * (4 - len(header_b64) % 4)  # Add padding
+            header = json.loads(base64.urlsafe_b64decode(header_b64))
+            kid = header.get("kid")
+        except Exception as e:
+            raise ValueError(f"Failed to extract key ID from token: {e}") from e
```

Ruff (0.14.11)

Avoid specifying long messages outside the exception class

(TRY003)